CRM/Contact/BAO/Query.php

```diff
@@ -436,6 +442,11 @@ public function __construct(
       $this->_params = array();
     }
 
+    // Search by Primary location type ONLY if searchPrimaryDetailsOnly setting is on OR
+    //  search query is trigerred via api OR
+    //  special parameter search_by_primary_details_only = TRUE
+    self::$_alwaysSearchByPrimaryOnly = (Civi::settings()->get('searchPrimaryDetailsOnly') || $apiEntity || !empty($params['search_by_primary_details_only']));
```

Comment:
@eileenmcnaughton didn't found the exact spot set ```search_by_primary_details_only = 1``` via all API params that uses location type. Reason why the earlier code used $apiEntity. Or can you suggest me we could set this parameter for API?